### ISSUE TITLE
Don't pollute $HOME

### ISSUE THREAD

**Issue by z0rc** [nylas/N1#674]
Please don't put your stuff into `~/.nylas`.

On OSX it should go under `~/Library/Application Support`.

On Linux you should use `$XDG_CONFIG_HOME`, `$XDG_CACHE_HOME` and `$XDG_DATA_HOME`, also with proper fallbacks to `~/.config`, `~/.cache` and `~/.local/share` if those variables are not set.

**Comment by connorjsmith** [comment]
+1

**Comment by bengotow** [comment]
Hey folks—there's a similar Atom thread here: https://github.com/atom/atom/issues/8281 (N1 started as a fork of Atom and inherited this convention). I definitely agree that there are more appropriate places, especially on Windows, where `.nylas` isn't even invisible. We'll look to fix this in a future release, but we need to investigate how it'd impact package management.

**Comment by noahbuscher** [comment]
:+1:

**Comment by grinich** [comment]
Note that Dropbox also does this on OS X (in `~/.dropbox`). We've mostly been looking for a first-order solution as we build out the app. Thanks for the feedback!

**Comment by mbilker** [comment]
Electron has a method [`app.getPath(name)`](https://github.com/atom/electron/blob/master/docs/api/app.md#appgetpathname) that includes OS-exclusive storage paths.

**Comment by danielbayley** [comment]
+1.

**Comment by mbilker** [comment]
For using `app.getPath(...)` from http://electron.atom.io/docs/v0.36.7/api/app/#appgetpathname

I tested this under Linux,
```
> require('remote').require('electron').app.getPath('userData')
"/home/mbilker/.config/Nylas N1"

> require('remote').require('electron').app.getPath('cache')
"/home/mbilker/.cache"
```

This would make N1 more compliant and put its data in more user-friendly folders.

**Comment by OJFord** [comment]
@mbilker How bizarre, then, that Atom [doesn't use it and even originally refused requests to implement XDG_CONFIG_HOME](https://github.com/atom/atom/issues/8281)!

**Comment by mbilker** [comment]
I agree that it is bizarre. N1 should switch before more plugins are made
to avoid any comparability issues. At least target 0.5.x

**Comment by caesar** [comment]
It would be great if you could change this. Stop polluting my home directory. There are standard places to store your config, data, and caches. The XDG spec has been around for 13 years now after all; there's no reason for modern apps to ignore it. (And as the original reporter mentioned, there are also standard places on OS X and Windows – but even if you wanted to use XDG conventions across all platforms that would be a *start*.)

**Comment by danielbayley** [comment]
> Stop polluting my home directory.

This!

Respect `XDG_` and also `NYLAS_HOME` should be implemented **_everywhere_** `~/.nylas` is hardcoded.

**Comment by miedzinski** [comment]
Did a fresh install of Nylas Mail 2.0. Still pollutes `$HOME` with `.nylas-mail` directory, despite having all XDG dirs configured.
